libs/blocks/editorial-card/editorial-card.js

```diff
@@ -0,0 +1,121 @@
+import { createTag, loadStyle, getConfig } from '../../utils/utils.js';
+import { decorateBlockBg, decorateBlockText, decorateBlockHrs, decorateTextOverrides, applyHoverPlay } from '../../utils/decorate.js';
+
+function decorateLockupFromContent(el) {
+  const rows = el.querySelectorAll(':scope > div > p');
+  const firstRowImg = rows[0]?.querySelector('img');
+  if (firstRowImg) {
+    rows[0].classList.add('lockup-area');
+    rows[0].childNodes.forEach((node) => {
+      if (node.nodeType === 3 && node.nodeValue !== ' ') {
+        const newSpan = createTag('span', { class: 'lockup-label' }, node.nodeValue);
+        node.parentElement.replaceChild(newSpan, node);
+      }
+    });
+  }
+}
+
+const extendDeviceContent = (el) => {
+  // check if there is a .body- above the .detail-
+  const detail = el.querySelector('[class^="detail-"]');
+  const prevElem = detail?.previousElementSibling;
+  if (prevElem) {
+    const prevIsBody = Array.from(prevElem.classList).some((c) => c.startsWith('body-'));
+    if (!prevIsBody) return;
+    prevElem.classList.remove('body-m');
+    prevElem.classList.add('body-xxs', 'device');
+  }
+};
+
+const decorateForeground = (el, rows, media = 0) => {
+  if (media) {
+    media.classList.add('media-area');
+    const mediaVideo = media.querySelector('video');
+    if (mediaVideo) {
+      applyHoverPlay(mediaVideo);
+    }
+    if (media.children.length > 1) decorateBlockBg(el, media);
+  }
+  rows.forEach((row, i) => {
+    if (i === 0) {
+      row.classList.add('foreground');
+      decorateLockupFromContent(row);
+    } else if (i === (rows.length - 1)) {
+      row.classList.add('footer');
+    } else {
+      row.classList.add('static');
+    }
+    decorateBlockText(row, ['m', 'm', 'm']); // heading, body, detail
+    decorateBlockHrs(row);
+  });
+};
+
+const decorateBgRow = (el, row, node) => {
+  const bgEmpty = row.textContent.trim() === '';
+  if (bgEmpty) {
+    el.classList.add('no-bg');
+    row.remove();
+  } else {
+    decorateBlockBg(el, node);
+  }
+};
```

Comment:
row & node is the same thing, right?

```suggestion
const decorateBgRow = (el, background) => {
  if (background.textContent.trim() === '') {
    el.classList.add('no-bg');
    row.remove();
    return
  } 
  decorateBlockBg(el, background);
};
```

Reply:
You're right! Thanks. This is updated/tested